### PR TITLE
lighting docs

### PR DESCRIPTION
**Pull request by jianhuang01**
For #835

#### Change List
- Docs for new lighting and material classes

### CODE REVIEW

File: docs/api-reference/material/phong-material.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+## Methods
+
+### constructor
+
+The constructor for the `PhongMaterial` class. Use this to create a new `PhongMaterial`.
+
+```js
+const phongMaterial = new PhongMaterial({ambient, diffuse, shininess, specularColor});
+```
+#### Parameters
```
Comment (ibgreen) [inline docs/api-reference/material/phong-material.md]: Drop the `####`

File: docs/api-reference/material/phong-material.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# PhongMaterial
+
+A material class specifies reflection properties of a shinny surface, uses `Blinn-Phong` model for underlying implementation.
```
[the remark this answers is not in the document]
Comment (ibgreen) [reply docs/api-reference/material/phong-material.md]: Explain the parameters in words:

In this material model, a material is characterized by ...

File: modules/core/src/materials/phong-material.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const defaultProps = {
 
 export default class PhongMaterial extends Material {
   constructor(props) {
-    super(props);
+    super();
```
Comment (ibgreen) [inline modules/core/src/materials/phong-material.js]: Hmm... This looks a bit weird. Why not forward the props? I think we will need to restore this once we put some props in the base class.

**Comment by Pessimistress** [comment]
I suggest add a master doc for lighting in Developer Guide that
- explain key concepts (light source, material, etc.)
- contains complete code examples
- links to subclasses

File: docs/api-reference/lighting/ambient-light.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+The constructor for the `AmbientLight` class. Use this to create a new `AmbientLight`.
+
+```js
+const ambientLight = new AmbientLight({color, intensity});
+```
+#### Parameters
+
+* `color` - (*array*,)  RGB color of ambient light source, default value is `[255, 255, 255]`.
+* `intensity` - (*number*) Strength of ambient light source, default value is `1.0`.
```
[the remark this answers is not in the document]
Comment (jianhuang01) [reply docs/api-reference/lighting/ambient-light.md]: Better to explain the formula in a master lighting doc, will have that when we have a scene class in luma.gl, so I can add example code there, too.

**Pull request merged**